### PR TITLE
Unwrapping soft exception in aspectJ

### PR DESCRIPTION
**Pull request by balalakshmifk**
Fixes https://github.com/flipkart-incubator/gojira/issues/57

### CODE REVIEW

File: modules/gojira-core/src/test/java/com/flipkart/gojira/core/AspectjMethodInterceptionTest.java [hunk 1]
```diff
@@ -49,4 +55,9 @@ public void testMethodInterception_nonAnnotated () throws Throwable {
         assertEquals(11, testNonAnnotated(10));
     }
 
+    @Test
+    public void testMethodInteception_exception () throws Throwable {
```
Comment (pmohankumar) [inline modules/gojira-core/src/test/java/com/flipkart/gojira/core/AspectjMethodInterceptionTest.java:59]: typo. and why snake_case?

File: modules/gojira-core/src/main/java/com/flipkart/gojira/core/aspects/AOPAllianceAdapter.java [hunk 2]
```diff
@@ -50,11 +50,7 @@ public Object around(ProceedingJoinPoint thisJoinPoint) throws Throwable {
         };
         MethodInterceptor mInt = getMethodInterceptor();
         if (mInt != null) {
-            try {
-                return mInt.invoke(mic);
-            } catch (Throwable t) {
-                throw new SoftException(t);
-            }
+            return mInt.invoke(mic);
```
Comment (pmohankumar) [inline modules/gojira-core/src/main/java/com/flipkart/gojira/core/aspects/AOPAllianceAdapter.java:53]: Let's remove the import of SoftException as well

Comment (balalakshmifk) [reply modules/gojira-core/src/main/java/com/flipkart/gojira/core/aspects/AOPAllianceAdapter.java:52]: Done

File: modules/gojira-core/src/test/java/com/flipkart/gojira/core/AspectjMethodInterceptionTest.java (hunk 1 above)
Comment (balalakshmifk) [reply modules/gojira-core/src/test/java/com/flipkart/gojira/core/AspectjMethodInterceptionTest.java]: Done

**Pull request merged**